### PR TITLE
Add config options from env for BatchSpanProcessor

### PR DESCRIPTION
**Pull request by TommyCpp**
part of #168

### CODE REVIEW

File: src/sdk/trace/span_processor.rs [hunk 1]
```diff
@@ -269,6 +283,56 @@ impl BatchSpanProcessor {
...
+    ) -> BatchSpanProcessorBuilder<E, S, I>
+        where
+            E: exporter::trace::SpanExporter,
+            S: Fn(BatchSpanProcessorWorker) -> SO,
+            I: Fn(time::Duration) -> IO,
+    {
+        let mut config = BatchConfig::default();
+        let schedule_delay = u64::from_str(
+            std::env::var(OTEL_BSP_SCHEDULE_DELAY_MILLIS)
+                .unwrap_or_else(|_| "".to_string())
```
Comment (jtescher) [inline src/sdk/trace/span_processor.rs:301]: Would these be simpler as:

```rust
std::env::var(OTEL_BSP_SCHEDULE_DELAY_MILLIS)
    .and_then(|delay| u64::from_str(&delay).ok())
    .unwrap_or(OTEL_BSP_SCHEDULE_DELAY_MILLIS_DEFAULT);
```

At a quick glance it's not clear that `"".to_string()` isn't 0, or some other case.

File: src/sdk/trace/span_processor.rs [hunk 2]
```diff
@@ -269,6 +283,56 @@ impl BatchSpanProcessor {
             config: Default::default(),
         }
     }
+
+    /// Create a new batch processor builder and set the config value based on environment variables.
+    pub fn from_env<E, S, SO, I, IO>(
```
Comment (jtescher) [inline src/sdk/trace/span_processor.rs:288]: Do you prefer the explicit `from_env` here? Has advantages of being clear when it is looking from env, alternative could be to have the batch config be options and fall back to checking env variables for any unset values sorta merging both constructors?

**Review by jtescher — COMMENTED** [review]
Great start @TommyCpp, had a couple of questions but looking good

File: src/sdk/trace/span_processor.rs (hunk 2 above)
Comment (TommyCpp) [reply src/sdk/trace/span_processor.rs:288]: I guess that depends on whether we should assume users always want to apply environment variable configuration. Personally I'd like know exactly where does the configuration comes from. But we could use environment variable to fill unset values in configs too.

Comment (jtescher) [reply src/sdk/trace/span_processor.rs:288]: Yeah I see the pros and cons of both (e.g. the jaeger exporter has an explicit `from_env`). I think it is fine for now to get feedback from the community on which is preferred.

File: src/sdk/trace/span_processor.rs [hunk 3]
```diff
@@ -269,6 +283,57 @@ impl BatchSpanProcessor {
...
+    ) -> BatchSpanProcessorBuilder<E, S, I>
+        where
+            E: exporter::trace::SpanExporter,
+            S: Fn(BatchSpanProcessorWorker) -> SO,
+            I: Fn(time::Duration) -> IO,
+    {
+        let mut config = BatchConfig::default();
+        let schedule_delay = std::env::var(OTEL_BSP_SCHEDULE_DELAY_MILLIS)
+            .and_then(|delay| u64::from_str(&delay).or(Ok(OTEL_BSP_SCHEDULE_DELAY_MILLIS_DEFAULT)))
+            .unwrap();
```
Comment (jtescher) [inline src/sdk/trace/span_processor.rs:307]: This way will panic if the var is not set I think? Probably best to go with a version that doesn't have a straight `.unwrap` in it, those tend to make people nervous when they look at the code.

**Review by jtescher — CHANGES_REQUESTED** [review]

File: src/sdk/trace/span_processor.rs (hunk 3 above)
Comment (TommyCpp) [reply src/sdk/trace/span_processor.rs:307]: Yeah you are right. Should have tested it with no env vars set I guess. Fixed it.

**Review by jtescher — APPROVED** [review]
Thanks @TommyCpp

**Pull request merged**